### PR TITLE
CLI: add type annotations in javascript main config files

### PR DESCRIPTION
**Pull request by yannbf**
Issue: N/A

## What I did

I realized we could get type annotations in javascript for main.js too (thanks @JReinhold)

so this PR changes the generated main.js files to:
```js
/** @type { import('@storybook/react-vite').StorybookConfig } */
export default {
  "stories": [
    "../src/**/*.mdx",
    "../src/**/*.stories.@(js|jsx|ts|tsx)"
  ],
};
```

It also adds tests to the file, which had none so far

## How to test

1. build the cli
2. go to a js project without Storybook
3. <path/to/storybook>/code/lib/cli/index.js init
4. Assess the resulting main.js

## Checklist

- [x] Make sure your changes are tested (stories and/or unit, integration, or end-to-end tests)
- [ ] Make sure to add/update documentation regarding your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

- [ ] If this PR should be tested against many or all sandboxes,
      make sure to add the `ci:merged` or `ci:daily` GH label to it.
- [x] Make sure this PR contains **one** of the labels below.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

### CODE REVIEW

**Comment by yannbf** [comment]
Open question:

Given that in Typescript we generate this:
```ts
import { StorybookConfig from } '@storybook/react-vite'
const config: StorybookConfig {
  "stories": [
    "../src/**/*.mdx",
    "../src/**/*.stories.@(js|jsx|ts|tsx)"
  ],
};
export default config
```

We could potentially make js look like this and keep the files more consistent:
```js
/** @type { import('@storybook/react-vite').StorybookConfig } */
const config = {
  "stories": [
    "../src/**/*.mdx",
    "../src/**/*.stories.@(js|jsx|ts|tsx)"
  ],
};
export default config
```

and probably in the docs we follow a similar pattern than for Typescript:
```js
// Replace your-framework with the framework you are using (e.g., react-webpack5, vue3-webpack5)
/** @type { import('@storybook/your-framework).StorybookConfig } */
const config = {
  "stories": [
    "../src/**/*.mdx",
    "../src/**/*.stories.@(js|jsx|ts|tsx)"
  ],
};
export default config
```

WDYT @kasperpeulen @jonniebigodes @shilman ?

**Comment by shilman** [comment]
@yannbf That's really cool!! ❤️

**Comment by yannbf** [comment]
> @yannbf That's really cool!! ❤️
> 
> To clarify: you're not saying get rid of the generated TS in CLI or documentation, but simply how to improve the JS side of things, right?

Yup, for sure!

**Comment by jonniebigodes** [comment]
This is amazing @yannbf 🤩 ! If I understand this correctly, I think we can probably shrink down the number of code snippet files we have in our documentation. I'll follow up with you on it, and we'll go from there. Thank you both for thinking about this (@yannbf and @JReinhold ). Appreciate it 🙏

**Comment by kasperpeulen** [comment]
I think this is great!

@jonniebigodes I don't think it would not reduce the code snippets, but make them more consistent right @yannbf ?

On the other hand, we should really just generate the JS examples from the TS examples with a codemod! @shilman What do you think, should be achievable right?

File: code/lib/cli/src/generators/configure.test.ts [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        name: '@storybook/react-vite',
+      },
+    });
+
+    const { calls } = (fse.writeFile as unknown as jest.Mock).mock;
+    const [mainConfigPath, mainConfigContent] = calls[0];
+
+    expect(mainConfigPath).toEqual('./.storybook/main.ts');
+    expect(mainConfigContent).toMatchInlineSnapshot(`
+      "import { StorybookConfig } from '@storybook/react-vite';
```
Comment (kasperpeulen) [inline code/lib/cli/src/generators/configure.test.ts:57]: I think better `import type`

**Comment by shilman** [comment]
Interesting idea @kasperpeulen -- I like it as a post 7.0 task

File: code/lib/cli/src/generators/configure.test.ts (hunk 1 above)
Comment (yannbf) [reply code/lib/cli/src/generators/configure.test.ts]: Done!
